src/backend/distributed/commands/table.c

```diff
@@ -507,6 +510,16 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		return PreprocessAlterTableStmt(node, alterTableCommand, processUtilityContext);
 	}
 
+	if (AlterTableDropsForeignKey(alterTableStatement))
```

Comment:
We are in the `PreprocessAlterTableStmt`, so what if the command fails on utility hook? I guess that's OK for now. @onurctirtir any thoughts?

Reply:
since we use `MarkInvalidateForeignKeyGraph` instead of directly invalidating the graph, we should be fine since such commands without `CASCADE` would already fail and I think we won't invalidate foreign key graph

Reply:
It will invalidate as we call `InvalidateForeignKeyGraphForDDL` in the catch block as well. But should be fine, just some overhead.

Still, might be good to comment that we are optimistically marking the graph invalid, should be OK even if this command fails on process utility